src/jit/decomposelongs.cpp

```diff
@@ -113,8 +113,6 @@ void DecomposeLongs::DecomposeRangeHelper()
     {
         node = DecomposeNode(node);
     }
-
-    assert(Range().CheckLIR(m_compiler, true));
```

Comment:
Why was this removed?

Reply:
We have two phases: DecomposeBlock and LowerBlock. This check was called between them.
The contained flag doesn't have to be correctly set before we call `LowerBlock()`. 
We have an assert that `GT_LONG` is contained or doesn't have a user in this `CheckLIR`.
So this check failed on x86.

Reply:
Interesting. It would be nice if we could leave this in, as it has caught problems before. Where specifically is the check you mention?

Reply:
It is in `isContained`:
```
#if !defined(_TARGET_64BIT_)
    if (OperGet() == GT_LONG)
    {
        // GT_LONG nodes are normally contained. The only exception is when the result
        // of a TYP_LONG operation is not used and this can only happen if the GT_LONG
        // has no parent.
        assert(isMarkedContained || (gtGetParent(nullptr) == nullptr));
    }
#endif
```
This check is useful so it seems unlikely to skip it when we call CheckLIR from the intermediate state.

Reply:
I think that we should move this check out of `isContained` and into whatever phase expects this behavior (likely codegen). This should be simple: there are already cases for `GT_LONG` in codegen{x,arm}arch.cpp, so the requisite assert (namely, that an uncontained `GT_LONG` encountered during codegen has a user) could easily be added there. This would clarify the requirement a bit by making it obvious which phase expects this behavior and would allow us to keep checking LIR correctness between decomp and lowering.

Reply:
It sounds reasonable. 
`genConsumeRegs` is the place, where we expect this flag to be set.
I updated PR with the fix.